scripts/perf/dags/elastic_dag.py

```diff
@@ -74,14 +74,78 @@ def safe_dag_id(s: str) -> str:
     return re.sub('[^0-9a-zA-Z_]+', '_', s)
 
 
-# TODO: We should add more shape types e.g. binary tree
+def chain_as_binary_tree(*tasks: BashOperator):
+    r'''
+    Chain tasks as a binary tree where task i is child of task (i - 1) // 2 :
+
+        t0 -> t1 -> t3 -> t7
+          |    \
+          |      -> t4 -> t8
+          |
+           -> t2 -> t5 -> t9
+               \
+                 -> t6
+    '''
+    for i in range(1, len(tasks)):
+        tasks[i].set_downstream(tasks[(i - 1) // 2])
+
+
+def chain_as_grid(*tasks: BashOperator):
+    '''
+    Chain tasks as a grid:
+
+     t0 -> t1 -> t2 -> t3
+      |     |     |
+      v     v     v
+     t4 -> t5 -> t6
+      |     |
+      v     v
+     t7 -> t8
+      |
+      v
+     t9
+    '''
+    if len(tasks) > 100 * 99 / 2:
+        raise ValueError('Cannot generate grid DAGs with lateral size larger than 100 tasks.')
+    grid_size = min([n for n in range(100) if n * (n + 1) / 2 >= len(tasks)])
+
+    def index(i, j):
+        '''
+        Return the index of node (i, j) on the grid.
+        '''
+        return int(grid_size * i - i * (i - 1) / 2 + j)
+
+    for i in range(grid_size - 1):
+        for j in range(grid_size - i - 1):
+            if index(i + 1, j) < len(tasks):
+                tasks[index(i + 1, j)].set_downstream(tasks[index(i, j)])
+            if index(i, j + 1) < len(tasks):
+                tasks[index(i, j + 1)].set_downstream(tasks[index(i, j)])
+
+
+def chain_as_star(*tasks: BashOperator):
+    '''
+    Chain tasks as a star (all tasks are children of task 0)
+
+     t0 -> t1
+      | -> t2
+      | -> t3
+      | -> t4
+      | -> t5
+    '''
+    tasks[0].set_upstream(tasks[1:])
```

Comment:
```suggestion
    tasks[0].set_upstream(list(tasks[1:]))
```
This should please mypy. Or we can change sigature of `set_upstream` to accept any of (list, tuple, set).

Reply:
just pushed your suggestion 👍
I do not have the experience to judge whether changing the signature of `set_upstream` is appropriate so I'll let you judge of that.